config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java

```diff
@@ -41,49 +41,38 @@ public FilesApplicationFile(Path path, File file) {
     }
 
     @Override
-    public boolean isDirectory() {
-        return file.isDirectory();
-    }
+    public boolean isDirectory() { return file.isDirectory(); }
 
     @Override
-    public boolean exists() {
-        return file.exists();
-    }
+    public boolean exists() { return file.exists(); }
 
     @Override
     public ApplicationFile delete() {
-        log.log(Level.FINE, () -> "Delete " + file);
-        if (file.isDirectory() && !listFiles().isEmpty()) {
-            throw new RuntimeException("files. Can't delete, directory not empty: " + this  + "(" + listFiles() + ")." + listFiles().size());
-        }
-        if (file.isDirectory() && file.listFiles() != null && file.listFiles().length > 0) {
-            for (File f : file.listFiles()) {
-                deleteFile(f);
+        if (file.isDirectory()) {
+            if (!listFiles().isEmpty())
```

Comment:
I believe this `delete()` can only be invoked on a directory if the directory only has entries that start with `.`, and in that case any such file and directory-trees are deleted.

This seems strange to me - does it make sense to you?

It is possibly related to the fact that this class creates a .meta directory after deletion.

Reply:
Discussed offline, added some TODOs